src/main/java/org/opensearch/security/resources/ResourceActionGroupsHelper.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.resources;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.security.securityconf.FlattenedActionGroups;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.ActionGroupsV7;
+
+import org.yaml.snakeyaml.Yaml;
+
+// CS-SUPPRESS-SINGLE: RegexpSingleline get Resource Sharing Extensions
+/**
+ * Helper class to load `resource-action-groups.yml` file for all resource sharing extensions.
+ */
+public class ResourceActionGroupsHelper {
+    public static final Logger log = LogManager.getLogger(ResourceActionGroupsHelper.class);
+
+    /**
+     * Loads action-groups config from the {@code resource-action-groups.yml} file from each resource sharing extension
+     * @param resourcePluginInfo will store the loaded action-groups config
+     *
+     * Sample yml file:
+     *   resource_types:
+     *   org.opensearch.sample.SampleResource:
+     *       sample_read_only:
+     *         - "cluster:admin/sample-resource-plugin/get"
+     *         - "indices:data/read*"
+     */
+    public static void loadActionGroupsConfig(ResourcePluginInfo resourcePluginInfo) {
```

Comment:
I think actionGroup makes sense for the backend, but on the UI I'd like to refer to these as Access Levels. They are not mutually exclusive and different levels should progressively give more access.

i.e. in the Quip use-case the lowest level of access would be read, then read + comment, then read + comment, edit and the highest-level of access would be full access which is read, comment, edit + share

Reply:
we definitely should.. In next version of access-levels for this feature?